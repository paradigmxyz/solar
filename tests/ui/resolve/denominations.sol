contract C {
    uint256 public a = 0x123 ether; //~ ERROR: cannot be used with unit denominations
    uint256 public b = 0x123 days; //~ ERROR: cannot be used with unit denominations
    uint256 public h = 1 years; //~ ERROR: unit denomination is deprecated

    // OK
    uint256 public i = 1 seconds;
    uint256 public j = 1 minutes;
    uint256 public k = 1 hours;
    uint256 public l = 1 days;
    uint256 public m = 1 weeks;
}
