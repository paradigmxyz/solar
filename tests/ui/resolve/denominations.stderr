error: hexadecimal numbers cannot be used with unit denominations
  --> ROOT/tests/ui/resolve/denominations.sol:LL:CC
   |
LL |     uint256 public a = 0x123 ether;
   |                        ^^^^^
   |
   = help: you can use an expression of the form "0x1234 * 1 days" instead

error: hexadecimal numbers cannot be used with unit denominations
  --> ROOT/tests/ui/resolve/denominations.sol:LL:CC
   |
LL |     uint256 public b = 0x123 days;
   |                        ^^^^^
   |
   = help: you can use an expression of the form "0x1234 * 1 days" instead

error: using "years" as a unit denomination is deprecated
  --> ROOT/tests/ui/resolve/denominations.sol:LL:CC
   |
LL |     uint256 public h = 1 years;
   |                        ^
   |

error: aborting due to 3 previous errors

